network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractIdentifiableImpl.java

```diff
@@ -264,7 +264,12 @@ public boolean removeProperty(String key) {
         Resource<D> r = getResource();
         Map<String, String> properties = r.getAttributes().getProperties();
         if (properties != null) {
-            return properties.remove(key) != null;
+            String oldValue = properties.get(key);
+            if (properties.remove(key) != null) {
+                index.updateResource(resource);
```

Comment:
You have to use the updateResource(lambda) and not use directly the index.updateResource